main.js

```diff
@@ -0,0 +1,6062 @@
+      "statusType": "NORMAL",
+      "createdAt": "2018-08-12T10:31:55Z"
+    },
+    {
+      "id": "085103ee-c597-49dc-ad6a-7fcc3211e4e1",
+      "assetId": "51f135d2-2648-4eb0-bc8e-5727b7cd6f13",
+      "statusType": "NORMAL",
+      "createdAt": "2018-08-15T16:31:22Z"
+    },
+    {
+      "id": "dafd7d07-b254-43c1-b899-b64b2f81c2ff",
+      "assetId": "4f436a0d-903b-46bd-8e45-46c1dd3dc631",
+      "statusType": "NORMAL",
+      "createdAt": "2018-08-16T21:48:00Z"
+    },
+    {
+      "id": "b70b56b0-fffc-48d1-ad54-f17a088927c9",
+      "assetId": "51f135d2-2648-4eb0-bc8e-5727b7cd6f13",
+      "statusType": "NORMAL",
+      "createdAt": "2018-08-19T05:12:42Z"
+    },
+    {
+      "id": "9fe17653-44d0-4307-9508-7c7822bd360e",
+      "assetId": "4d642eef-ff1e-4c4a-9d17-318dc696a90c",
+      "statusType": "NORMAL",
+      "createdAt": "2018-08-19T10:28:29Z"
+    },
+    {
+      "id": "e90c785a-8918-4ccd-9dec-9c99dd346c98",
+      "assetId": "62d98170-ff7e-4ee1-8f1c-b54f8aed351c",
+      "statusType": "WARNING",
+      "createdAt": "2018-08-20T02:18:36Z"
+    },
+    {
+      "id": "84d5aa27-1476-40ba-9689-63642b36464a",
+      "assetId": "8fded256-979f-402c-aacc-8ab6b98787d1",
+      "statusType": "WARNING",
+      "createdAt": "2018-08-21T19:50:06Z"
+    },
+    {
+      "id": "e9a5a5bf-e9c8-40fb-beb6-ad425dc792db",
+      "assetId": "b76782b6-4632-4144-83a4-efc874ba24d0",
+      "statusType": "NORMAL",
+      "createdAt": "2018-08-24T09:40:44Z"
+    },
+    {
+      "id": "ddb464fd-5223-429c-843f-720aaa4fcb8c",
+      "assetId": "d4250e33-f81f-408f-855a-36414fe4cee5",
+      "statusType": "WARNING",
+      "createdAt": "2018-08-24T17:24:39Z"
+    },
+    {
+      "id": "be8e18db-5eb2-4d19-94bd-de4864002a1d",
+      "assetId": "fbbcaf29-a72f-48f7-8f2b-d5abf8bc24bf",
+      "statusType": "NORMAL",
+      "createdAt": "2018-08-25T20:38:47Z"
+    },
+    {
+      "id": "bea124fd-b7ed-44bb-a370-76c5ae161f27",
+      "assetId": "4f436a0d-903b-46bd-8e45-46c1dd3dc631",
+      "statusType": "WARNING",
+      "createdAt": "2018-08-29T06:03:56Z"
+    },
+    {
+      "id": "a3954db4-7211-4005-837a-78aa405457eb",
+      "assetId": "08b86fa4-b2be-4005-93e9-35d08b4c17a7",
+      "statusType": "NORMAL",
+      "createdAt": "2018-08-30T10:20:42Z"
+    }
+  ];
+
+let assestState = {};
+
+
+
+function handle(message){
+    if(assestState[message.assetId]){
+      if(assestState[message.assetId][1] !== message.statusType && message.statusType !== "NORMAL"){
+       return console.warn('Assest '+ message.assetId + ' is about to fail  with status ' + message.statusType + ' at ' + new Date().toLocaleString())
+    }
+     if(assestState[message.assetId][1] !== "NORMAL" && message.statusType === 'NORMAL'){
+       return console.info('Assest '+ message.assetId + ' has recovered with status ' + message.statusType + ' at ' + new Date().toLocaleString())
+     }
+} 
+     assestState[message.assetId] = [message.assetId,message.statusType]

```

Comment:
Why do you need to store the `assetId` twice? What is the purpose? Is this useful for any use case?

You could simply do the following:
```suggestion
     assestState[message.assetId] = message.statusType
```